dispatch/static/manager/src/js/components/ContentEditor/embeds/VideoEmbed.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+
+import { FormInput, TextInput } from '../../inputs'
+
+class VideoEmbedComponent extends React.Component {
+
+  constructor(props){
+    super(props)
+
+    this.handleUrlChange = this.handleUrlChange.bind(this)
+    this.insertVideo = this.insertVideo.bind(this)
+
+    this.state = {
+      inserted: false,
+      validURL: false
+    }
+  }
+
+  handleUrlChange(e){
+    e.preventDefault()
+    this.props.updateField('url', e.target.value)
+    console.log("URL IS: "+this.props.data.url);
+
+  }
+
+
+  insertVideo(){
+    var id = this.validYouTube(this.props.data.url);
+    this.props.updateField('id',id)
+    console.log("ID IS: "+this.props.data.id)
+
+    if(this.props.data.id){
+      this.setState({
+        validURL: true,
+        inserted : true
+      })
+    }
+  }
+
+  validYouTube(url){
+     var p = /^(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?(?=.*v=((\w|-){11}))(?:\S+)?$/
+     return (url.match(p)) ? RegExp.$1 : false
```

Comment:
`RegExp.$1..9` is [nonstandard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/n) and [deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features#RegExp_Properties) notation, according to Mozilla, and refers to icky global state to boot.

It would be much more reliable to use the array returned by [`url.match(p)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match). For example:

```js
const p = ...
const [, videoId] = url.match(p) || []
return videoId